32.md

```diff
@@ -0,0 +1,172 @@
+with the social graph in view to provide a better user experience.
+
+Label Target
+----
+
+The label event MUST include one or more tags representing the object or objects being
+labeled: `e`, `p`, `a`, `r`, or `t` tags. This allows for labeling of events, people, relays,
+or topics respectively. As with NIP-01, a relay hint SHOULD be included when using `e` and
+`p` tags.
+
+Label Tag
+----
+
+This NIP introduces a new tag `l` which denotes a label, and a new `L` tag which denotes a label namespace. A label MUST include a mark matching an `L` tag. `L` tags refer to a tag type within nostr, or a nomenclature external to nostr defined either formally or by convention. Any string can be a namespace, but publishers SHOULD ensure they are unambiguous by using a well-defined ISO standard or reverse domain name notation. Some examples:
+
+- `["l", "footstr", "#t"]` - the publisher thinks the given entity should have the `footstr` topic applied.
+- `["l", "<pubkey>", "#p"]` - the publisher things the given entity should be tagged with with `<pubkey>`
+- `["l", "D005528", "MeSH"]` - ["Foot"](https://meshb.nlm.nih.gov/record/ui?ui=D005528) from NIH's Medical Subject Headings vocabulary
+- `["l", "3173435", "GeoNames"]` - [Milan, Italy](https://www.geonames.org/3173435/milan.html) using the GeoNames coding system
+- `["l", "IT-MI", "ISO-3166-2"]` - Milano, Italy using ISO 3166-2.
+- `["l", "VI-hum", "social.nos.ontology"]` - Violence toward a human being as defined by ontology.nos.social.
+- `["l", "review", "social.coracle.ontology"]` - the publisher is leaving a review about the target, as defined by ontology.coracle.social.
+
+`L` tags containing the label namespaces MUST be included in order to support searching by
+namespace rather than by a specific tag. The special `ugc` ("user generated content") namespace
+MAY be used when the label content is provided by an end user.
+
+`l` and `L` tags MAY be added to other event kinds to support self-reporting. For events
+with a kind other than 1985, labels refer to the event itself.
+
+Label Annotations
+-----
+
+A label tag MAY include a 4th positional element detailing extra metadata about the label in question. This string
+should be a json-encoded object. Any key MAY be used, but the following are recommended:
+
+- `quality` may have a value of 0 to 1. This allows for an absolute, granular scale that can be represented in any way (5 stars, color scale, etc).
+- `confidence` may have a value of 0 to 1. This indicates the certainty which the author has about their rating.
+- `context` may be a comma-separated list of urls (including NIP-21 urls) indicating other context that should be considered when interpreting labels.
+
+Content
+-------
+
+Labels should be short, meaningful strings. Longer discussions, such as for a review, or an
+explanation of why something was labeled the way it was, should go in the event's `content` field.
+
```

Comment:
```suggestion
Label Deletion
---------------

Clients that support labels SHOULD also support their deletion via [NIP-09](https://github.com/nostr-protocol/nips/blob/master/09.md) events. Labels added at the time of publishing can only be deleted by deleting the original event.
```

Reply:
Maybe this is superfluous? But given all the debate between using replaceable events for labeling maybe it's worth calling out? I could go either way.

Reply:
I think it is, this is just the way events work.